client/clienterror/clienterror.go

```diff
@@ -333,6 +333,31 @@ func New(err error) *APIError {
 		}
 	}
 
+	// delete node pool
+	if deleteNodePoolsUnauthorizedErr, ok := err.(*nodepools.DeleteNodePoolUnauthorized); ok {
+		return &APIError{
+			HTTPStatusCode: http.StatusUnauthorized,
+			OriginalError:  deleteNodePoolsUnauthorizedErr,
+			ErrorMessage:   "Unauthorized",
+			ErrorDetails:   "You don't have permission to delete this node pool.",
+		}
+	}
+	if deleteNodePoolsNotFoundErr, ok := err.(*nodepools.DeleteNodePoolNotFound); ok {
+		return &APIError{
+			HTTPStatusCode: http.StatusNotFound,
+			OriginalError:  deleteNodePoolsNotFoundErr,
+			ErrorMessage:   "Not found",
+			ErrorDetails:   "The cluster or node pool was not found or you don't have access to it.",
```

Comment:
The other errors exclusively talk about node pools. This here mentions clusters as well. Is this how it should be?

Reply:
The thinking is that a 404 error could mean either that the node pool does not exist OR that the owning cluster does not exist. We don't know.